src/ape/managers/chain.py

```diff
@@ -185,8 +185,13 @@ def range(
         # Note: the range `stop_block` is a non-inclusive stop, while the
         #       `.query` method uses an inclusive stop, so we must adjust downwards.
         results = self.query("*", start_block=start, stop_block=stop - 1)  # type: ignore
```

Comment:
Curious if we add step to the block query type, and then that allows queries to deselect a lot of unused information when querying larger sets of blocks (say, every thousand blocks)

Reply:
Ah, yeah this probably makes sense at the query level